internal/datacoord/task_index.go

```diff
@@ -128,6 +129,13 @@ func (it *indexBuildTask) PreCheck(ctx context.Context, dependency *taskSchedule
 					FieldType: int32(partitionKeyField.DataType),
 					DataIds:   getBinLogIDs(segment, partitionKeyField.FieldID),
 				})
+				iso, _, isoErr := common.IsPartitionKeyIsolationPropEnabled(collInfo.Properties)
+				if isoErr != nil {
+					log.Ctx(ctx).Warn("failed to parse partition key isolation", zap.Error(isoErr))
```

Comment:
This parse error should not happen here. It should be checked way before it got set. If it indeed happens, not blocking index build.